### PR TITLE
Dockerfile for docs

### PR DESCRIPTION
**Pull request by TriplEight**
Parity-ethereum build image:
- new notifier image (official nightly)
- corrected descriptions (is it ok I mention private repo here?)
- beta and nightly Rust added to reintroduce beta and nightly tests
- new cache location to support shared runners
- removed docks dependencies

Parity-ethereum docs image:
- appears

All images will fail during run if errors occurred.

### CODE REVIEW

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# install tools and dependencies
+RUN set -eux; \
+	apt-get -y update; \
+    apt-get install -y --no-install-recommends \
+		curl ca-certificates git; \
+# install nodejs and yarn
+	curl -sL https://deb.nodesource.com/setup_10.x | bash -; \
+	curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -; \
+	echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list; \
+	apt-get -y update; \
```
Comment (gabreal) [inline docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile]: is it intentional to run it twice?

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile [hunk 2]
```diff
@@ -8,43 +9,46 @@ FROM ubuntu:xenial as builder
...
 
 # compiler ENV
 ENV CC=gcc \
 	CXX=g++ \
 	CARGO_TARGET=x86_64-unknown-linux-gnu \
-	CARGO_HOME=/cargo \
 # cache handler
 	RUSTC_WRAPPER=sccache \
-	SCCACHE_DIR=/cargo/sccache \
+	CARGO_HOME=/ci-cache/parity-ethereum/cargo \
```
[the remark this answers is not in the document]
Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile]: first `CARGO_HOME` is just for Rust installation, to save it's `bin` there, it's not needed after.
the second one is for cache

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile (hunk 1 above)
Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile]: unfortunately yes, first time to be able to install anything, and this update is to fetch info from those new repositories

Comment (gabreal) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile]: why not add repos at the top?

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabreal) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile]: can it then be removed afterwards?

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile (hunk 1 above)
Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile]: there were no curl and certs before!

Comment (gabreal) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile]: you could put just the echo line on top - but maybe this small speedup will hurt legibility more

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile]: I'll try to optimize image later

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile (hunk 1 above)
Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile]: this image won't likely to be rebuilt often

Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-docs/Dockerfile]: but I understand, will think how to optimize

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile]: tested, unfortunately without the first declaration is necessary for all the Rust tools installation,
the second one is for storing cache

**Pull request merged**